### PR TITLE
Spike topic exchange

### PR DESCRIPTION
**Pull request by NickGrantONS**
# Motivation and Context
RH want to consume CaseCreated and UacUpdated events on different queues, using a topic exchange instead of a fanout.

# What has changed
We have changed from a fanout exchange to two queues (one for RH and one for Action Scheduler) to a topic exchange to three queues.

# How to test?
Run the acceptance tests on the branch called `spike-topic-exchange` along with the Action Scheduler on the branch `spike-topic-exchange`.

# Links
Trello: https://trello.com/c/LQzjwQc2

# Screenshots (if appropriate):

### CODE REVIEW

**Comment by NickGrantONS** [comment]
[SPIKE: Use topic exchange instead of fanout for Case & UAC events (0.5 day)](https://trello.com/c/LQzjwQc2/739-spike-use-topic-exchange-instead-of-fanout-for-case-uac-events-05-day)

File: src/main/java/uk/gov/ons/census/casesvc/service/CaseProcessor.java [hunk 1]
```diff
@@ -16,13 +16,14 @@
 public class CaseProcessor {
 
   private static final String SURVEY = "CENSUS";
+  private static final String CASE_UPDATE_ROUTING_KEY = "event.case.update";
```
Comment (davidmort) [inline src/main/java/uk/gov/ons/census/casesvc/service/CaseProcessor.java]: would this make more sense as we're emitting a CaseCreated event here?
```suggestion
  private static final String CASE_CREATE_ROUTING_KEY = "event.case.create";
```

File: src/main/java/uk/gov/ons/census/casesvc/service/CaseProcessor.java [hunk 2]
```diff
@@ -49,7 +50,8 @@ public void emitCaseCreatedEvent(Case caze) {
     responseManagementEvent.setEvent(event);
     responseManagementEvent.setPayload(payload);
 
-    rabbitTemplate.convertAndSend(emitCaseEventExchange, "", responseManagementEvent);
+    rabbitTemplate.convertAndSend(
+        outboundExchange, CASE_UPDATE_ROUTING_KEY, responseManagementEvent);
```
Comment (davidmort) [inline src/main/java/uk/gov/ons/census/casesvc/service/CaseProcessor.java]: same as above
```suggestion
        outboundExchange, CASE_CREATE_ROUTING_KEY, responseManagementEvent);
```

File: src/main/java/uk/gov/ons/census/casesvc/service/CaseProcessor.java (hunk 1 above)
Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/census/casesvc/service/CaseProcessor.java]: this is what's in the spec here: https://collaborate2.ons.gov.uk/confluence/display/CATD/RM+Event+Model

Has been agreed with Rob Smart, Phil Whiles & Ella Cook... so gonna mark this as resolved.

File: src/main/java/uk/gov/ons/census/casesvc/service/CaseProcessor.java (hunk 2 above)
Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/census/casesvc/service/CaseProcessor.java]: this is what's in the spec here: https://collaborate2.ons.gov.uk/confluence/display/CATD/RM+Event+Model

Has been agreed with Rob Smart, Phil Whiles & Ella Cook... so gonna mark this as resolved.

**Pull request merged**